aspnetcore/grpc/performance.md

```diff
@@ -153,6 +153,9 @@ var channel = GrpcChannel.ForAddress("https://localhost:5001", new GrpcChannelOp
 
 The preceding code configures a channel that sends a keep alive ping to the server every 60 seconds during periods of inactivity. The ping ensures the server and any proxies in use won't close the connection because of inactivity.
 
+> [!NOTE]
+> Keep alive pings only help keep the connection alive. Long-running gRPC calls on the connection may still be killed by the server or intermediary proxies for inactivity.
```

Comment:
>  Long-running gRPC calls on the connection may still be killed by the server

Could this link to documentation for how to configure the service-side per-call timeout (for a .NET-based gRPC server)?

Reply:
> > Long-running gRPC calls on the connection may still be killed by the server
> 
> Could this link to documentation for how to configure the service-side per-call timeout (for a .NET-based gRPC server)?

@chwarr  if you supply a link I can add it.

Reply:
I don't believe Kestrel has a limit by default.

Reply:
> @chwarr if you supply a link I can add it.

I don't know, which is why I asked. :-)

I gather from JameNK's reply that there is nothing in Kestrel nor gRPC.NET that will terminate an idle gRPC call (or a HTTP2 stream) on a HTTP2 connection that has keep alive enabled.

I was reading too much into phrasing like "Keep alive pings don't help to keep long-running gRPC calls alive." and "I don't believe Kestrel has a limit _by default_." (emphasis mine) and assuming that there was a Kestrel setting. It's entirely possible that App Service is doing something to the idle gRPC calls/stream, but I can't find any documentation about that either... :-(

Reply:
I think the App Service proxy has a maximum limit on request time based on some discussion I saw a while ago. I haven't tried it out myself.